sites/partners/src/listings/PaperListingDetails/sections/DetailUnits.tsx

```diff
@@ -1,47 +1,118 @@
 import React, { useContext, useMemo } from "react"
-import { t, GridSection, MinimalTable, Button, ViewItem } from "@bloom-housing/ui-components"
+import { t, GridSection, MinimalTable } from "@bloom-housing/ui-components"
 import { ListingContext } from "../../ListingContext"
 import { UnitDrawer } from "../DetailsUnitDrawer"
+import { MonthlyRentDeterminationType } from "@bloom-housing/backend-core/types"
 
 type DetailUnitsProps = {
   setUnitDrawer: (unit: UnitDrawer) => void
 }
 
+function isDefined(item: number | string) {
+  return item !== null && item !== undefined && item !== ""
+}
+
+function formatRange(min: string | number, max: string | number, prefix: string, postfix: string) {
+  if (!isDefined(min) && !isDefined(max)) return ""
+  if (min == max || !isDefined(max)) return `${prefix}${min}${postfix}`
+  if (!isDefined(min)) return `${prefix}${max}${postfix}`
+  return `${prefix}${min}${postfix} - ${prefix}${max}${postfix}`
+}
+
 const DetailUnits = ({ setUnitDrawer }: DetailUnitsProps) => {
   const listing = useContext(ListingContext)
 
   const unitTableHeaders = {
-    number: "listings.unit.number",
     unitType: "listings.unit.type",
-    amiPercentage: "listings.unit.ami",
-    monthlyRent: "listings.unit.rent",
-    sqFeet: "listings.unit.sqft",
-    priorityType: "listings.unit.priorityType",
-    status: "listings.unit.status",
+    units: "listings.unitsSummary.numUnits",
+    amiRange: "listings.unitsSummary.amiRange",
+    rentRange: "listings.unitsSummary.rentRange",
+    occupancyRange: "listings.unitsSummary.occupancyRange",
+    sqFeetRange: "listings.unitsSummary.sqftRange",
+    bathRange: "listings.unitsSummary.bathRange",
     action: "",
   }
 
+  const commaHelper = (rent, percent) => {
+    if (rent !== undefined && percent !== undefined) {
+      return ", "
+    }
+    return ""
+  }
+
   const unitTableData = useMemo(
     () =>
-      listing?.units.map((unit) => ({
-        number: unit.number,
-        unitType: unit.unitType && t(`listings.unitTypes.${unit.unitType.name}`),
-        amiPercentage: unit.amiPercentage,
-        monthlyRent: unit.monthlyRent,
-        sqFeet: unit.sqFeet,
-        priorityType: unit.priorityType?.name,
-        status: t(`listings.unit.statusOptions.${unit.status}`),
-        action: (
-          <Button
-            type="button"
-            className="font-semibold uppercase"
-            onClick={() => setUnitDrawer(unit)}
-            unstyled
-          >
-            {t("t.view")}
-          </Button>
-        ),
-      })),
+      listing?.unitGroups.map((unit) => {
+        const types = unit.unitType.map((type) => {
+          return t(`listings.unitsSummary.${type.name}`)
+        })
+        let amiRange = [undefined, undefined]
+        let rentRange = [undefined, undefined]
+        let percentIncomeRange = [undefined, undefined]
+        unit?.amiLevels?.forEach((ami) => {
+          if (ami.amiPercentage) {
```

Comment:
53-101 can by dry-ed up

Reply:
@YazeedLoonat , 
We had to do something similar in `units-transformations`, so created a function in `unit-transformation-helpers` (with tests even), however that uses the `MinMax` type. Given that we have that helper and a `MinMax` type, I think you can update this to take advantage of that